prometheus_juju_exporter/collector.py

```diff
@@ -125,20 +126,36 @@ def _get_labels_to_remove(self, gauge_name: str) -> None:
         for label in stale_labels.values():
             self.data[gauge_name]["labelvalues_remove"].append(list(label[0].values()))
 
-    def _get_machine_type(self, machine: Dict) -> MachineType:
+    def _get_machine_type(self, machine: Dict, machine_id: str) -> MachineType:
         """Detect machine type based on its MAC address.
 
         :param dict machine: status information for a machine
         :return MachineType: MachineType class object indicating the machine type
         """
-        interfaces = machine["network-interfaces"].values()
+        self.logger.debug("Determining type of machine: %s", machine_id)
+        machine_type = MachineType.METAL
+        skip_interfaces = self.config["detection"]["skip_interfaces"].as_str_seq()
+        interface_filter = re.compile(r"(?=(" + "|".join(skip_interfaces) + r"))")
```

Comment:
why is the lookahead needed?

Reply:
Truth be told, regexps are always confusing to me, so I googled around and this looked like it did the job. But you are right. I tested it now and 
```python
interface_filter = re.compile(r"|".join(skip_interfaces))
```
should do the trick, correct?

Reply:
I would expect it to, yes. The positive lookahead is useful for expressing "match this $THING if it is followed by this $OTHERTHING"; in your case you simply want "match one of these patterns"